external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/SparkAvroCallbackHandler.scala

```diff
@@ -53,7 +53,7 @@ private[flume] class SparkAvroCallbackHandler(val threads: Int, val channel: Cha
   // Since the new txn may not have the same sequence number we must guard against accidentally
   // committing a new transaction. To reduce the probability of that happening a random string is
   // prepended to the sequence number. Does not change for life of sink
-  private val seqBase = RandomStringUtils.randomAlphanumeric(8)
+  private val seqBase = UUID.randomUUID().toString.substring(0, 8)
```

Comment:
Last time, we tried it and we ended up getting the same string multiple times. I am not entirely sure why, but that was the reason we used RandomStringUtils in the first place.

The idea of using the seqBase is to protect against the case where a sink restart can cause collisions. I think this should be random enough, else we can use the full UUID which I feel is adding too much overhead without a whole lot of gain.